src/encoded/audit/genetic_modification.py

```diff
@@ -0,0 +1,31 @@
+from snovault import (
+    AuditFailure,
+    audit_checker,
+)
+
+# flag genetic modifications with purpose "tagging" that lack genetic modification characterization.
+def audit_tagging_modification(value, system):
+    if value['purpose'] == 'tagging' and\
+       not value.get('characterizations'):
+        detail = ('Genetic modification {} performed for the '
+                  'purpose of {} is missing validating characterization.').format(
+            value['@id'],
+            value['purpose']
+        )
+        yield AuditFailure(
+            'missing genetic modification characterization',
+            detail,
+            level='WARNING')
+        return
+
+
+function_dispatcher = {
+    'audit_tag': audit_tagging_modification,
+}
+
+@audit_checker('GeneticModification',
+               frame=['characterizations'])
+def audit_modification(value, system):
+    for function_name in function_dispatcher.keys():
```

Comment:
We don't actually use the key so use values() instead of keys(). 

The third choice it items() that returns key and value

Reply:
fixed